Snail/Observable.swift

```diff
@@ -100,11 +100,11 @@ public class Observable<T> : ObservableType {
         let scheduler = Scheduler(delay)
 
         var next: T?
-        scheduler.event.subscribe(onNext: {
-            guard let event = next else {
+        scheduler.observable.subscribe(onNext: {
+            guard let nextValue = next else {
```

Comment:
can we use `next` here too?

Reply:
Unfortunately no. I tried that originally, but the compiler needs to be able to differentiate between `nextValue` and `next` when it sets `next = nil`

Reply:
gotcha. In that case I think we should rename the other next to `nextValue` to be more clear, since it's really just the value part
```swift
var nextValue: T?
scheduler.observable.subscribe(onNext: {
    guard let nextValue = nextValue else {
        return
    }
    observable.on(.next(nextValue))
    nextValue = nil
})

subscribe(onNext: { next = $0 }, onError: { observable.on(.error($0)) }, onDone: { observable.on(.done) })
```

Reply:
Unless I'm misunderstanding what you're saying, I don't think that will work for the same reason. If both the unwrapped and wrapped variables have the same name, then the compiler complains at

```
nextValue = nil
```

that it can't change the value of a `let`, because it's trying to use the unwrapped variable.

Reply:
![image](https://user-images.githubusercontent.com/3113871/47164234-a5643c00-d2c5-11e8-9659-2d1532a1df0a.png)